packages/@aws-cdk/aws-kinesisfirehose/lib/delivery-stream.ts

```diff
@@ -0,0 +1,256 @@
+    if (!attrs.deliveryStreamName && !attrs.deliveryStreamArn) {
+      throw new Error('Either deliveryStreamName or deliveryStreamArn must be provided in DeliveryStreamAttributes');
+    }
+    const deliveryStreamName = attrs.deliveryStreamName ?? cdk.Stack.of(scope).parseArn(attrs.deliveryStreamArn!).resourceName;
+    if (!deliveryStreamName) {
+      throw new Error(`Could not import delivery stream from malformatted ARN ${attrs.deliveryStreamArn}: could not determine resource name`);
+    }
+    const deliveryStreamArn = attrs.deliveryStreamArn ?? cdk.Stack.of(scope).formatArn({
+      service: 'firehose',
+      resource: 'deliverystream',
+      resourceName: attrs.deliveryStreamName,
+    });
+    class Import extends DeliveryStreamBase {
+      public readonly deliveryStreamName = deliveryStreamName!;
+      public readonly deliveryStreamArn = deliveryStreamArn;
+      public readonly grantPrincipal = attrs.role ?? new iam.UnknownPrincipal({ resource: this });
+    }
+    return new Import(scope, id);
+  }
+
+  readonly deliveryStreamName: string;
+
+  readonly deliveryStreamArn: string;
+
+  readonly grantPrincipal: iam.IPrincipal;
+
+  constructor(scope: Construct, id: string, props: DeliveryStreamProps) {
+    super(scope, id);
+
+    const role = props.role ?? new iam.Role(this, 'Service Role', {
+      assumedBy: new iam.ServicePrincipal('firehose.amazonaws.com'),
+    });
+    this.grantPrincipal = role;
+
+    if (props.destinations.length !== 1) {
+      throw new Error(`Only one destination is allowed per delivery stream, given ${props.destinations.length}`);
+    }
+    const destinationConfig = props.destinations[0].bind(this, { deliveryStream: this });
+
+    const resource = new CfnDeliveryStream(this, 'Resource', {
+      deliveryStreamName: props.deliveryStreamName,
+      deliveryStreamType: 'DirectPut',
+      ...destinationConfig.properties,
+    });
+    resource.node.addDependency(this.grantPrincipal);
+
+    this.deliveryStreamArn = this.getResourceArnAttribute(resource.attrArn, {
+      service: 'kinesis',
+      resource: 'deliverystream',
+      resourceName: this.physicalName,
+    });
+    this.deliveryStreamName = this.getResourceNameAttribute(resource.ref);
+  }
+}
+
+function setConnections(scope: Construct) {
+  const region = cdk.Stack.of(scope).region;
+  let cidrBlock = RegionInfo.get(region).firehoseCidrBlock;
+  if (!cidrBlock) {
+    const mapping: {[region: string]: { FirehoseCidrBlock: string }} = {};
+    RegionInfo.regions.forEach((regionInfo) => {
+      if (regionInfo.firehoseCidrBlock) {
+        mapping[regionInfo.name] = {
+          FirehoseCidrBlock: regionInfo.firehoseCidrBlock,
+        };
+      }
+    });
+    const cfnMapping = new cdk.CfnMapping(scope, 'Firehose CIDR Mapping', {
+      mapping,
+    });
+    cidrBlock = cdk.Fn.findInMap(cfnMapping.logicalId, region, 'FirehoseCidrBlock');
```

Comment:
The solution to that would be to add the mapping at the top-level into the `Stack` and reuse it if already there.

Also, the logic of taking a value from `regioninfo` if possible and using a mapping otherwise is shared in a bunch of places. I think this is probably worth turning into a static function helper on `CfnMapping` to do this work (and we can put the reuse there).

Use a UUID to identify the mapping block or something.

Reply:
@BenChaimberg I'll let you take this one as well

Reply:
Opened #15617